pkg/registry/batch/job/strategy.go

```diff
@@ -265,6 +265,10 @@ func (jobStrategy) ValidateUpdate(ctx context.Context, obj, old runtime.Object)
 	oldJob := old.(*batch.Job)
 
 	opts := validationOptionsForJob(job, oldJob)
+	tempErrs := validation.ValidateJob(oldJob, opts)
+	if len(tempErrs) > 0 {
+		opts.AllowInvalidLabelValueInSelector = false
+	}
```

Comment:
compute this inside `validationOptionsForJob`, and only relax it if the validation errors were specifically due to an invalid label selector